lib/modules/hosts/giphy.js

```diff
@@ -1,27 +1,35 @@
+import { ajax } from '../../environment';
+import logo from '../../images/hosts/giphy-logo.png';
+
 export default {
 	moduleID: 'giphy',
 	name: 'giphy',
 	domains: ['giphy.com'],
-	logo: '//giphy.com/static/img/favicon.png',
-	detect: ({ pathname }) => (/^\/gifs\/(?:\w+-)*(.*?)(\/html5)?$/i).exec(pathname),
-	handleLink(href, [, id, isHtml5]) {
+	logo,
+	detect: ({ pathname }) => (/^(?:\/gifs|\/media|)\/(?:\w+-)*([^/.]*)(\/html5)?.*$/i).exec(pathname),
+	async handleLink(href, [, id, isHtml5]) {
+		const { data } = await ajax({
+			url: `http://api.giphy.com/v1/gifs/${id}?api_key=dc6zaTOxFJmzC`,
+			type: 'json',
+		});
+
 		if (isHtml5) {
```

Comment:
I'm fine with removing the special html5 logic. Based on the data returned from the giphy api, I didn't think it was necessary either but didn't know if there was another reason to keep it around.